src/SYS.ASM

```diff
@@ -69,7 +69,12 @@ main:                                           ; Main code for House-DOS kernel
 
     call data_kernseg                           ; Reset segment
 
+.new_command:
+    mov word [gs:pipe_offset], 0x0000           ; Reset command offset
+
 .handback:
+    call check_newline                          ; Check for a newline at the end of the buffer
```

Comment:
There should be a newline at the end of input/output for proper UNIX formatting (and to work properly in EDIT)